atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -2123,6 +2124,65 @@ public void compactInternally(TableReference tableRef) {
         }
     }
 
+    @Override
+    public NodeAvailabilityStatus getNodeAvailabilityStatus() {
+        if (!doesConfigReplicationFactorMatchWithCluster()) {
```

Comment:
Ah I got further down and saw that this indeed does all the actual checks and takes `safetyDisabled` into account.